fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biome/InternalBiomeData.java

```diff
@@ -109,97 +116,79 @@ public static void setOverworldRiverBiome(Biome primary, Biome river) {
+	public static Map<RegistryKey<Biome>, WeightedBiomePicker> getOverworldEdges() {
 		return OVERWORLD_EDGE_MAP;
 	}
 
-	public static Map<Biome, Biome> getOverworldRivers() {
+	public static Map<RegistryKey<Biome>, RegistryKey<Biome>> getOverworldRivers() {
 		return OVERWORLD_RIVER_MAP;
 	}
 
 	public static EnumMap<OverworldClimate, WeightedBiomePicker> getOverworldModdedContinentalBiomePickers() {
 		return OVERWORLD_MODDED_CONTINENTAL_BIOME_PICKERS;
 	}
 
-	public static Map<Biome, VariantTransformer> getOverworldVariantTransformers() {
+	public static Map<RegistryKey<Biome>, VariantTransformer> getOverworldVariantTransformers() {
 		return OVERWORLD_VARIANT_TRANSFORMERS;
 	}
 
-	public static Set<Biome> getNetherBiomes() {
+	public static Set<RegistryKey<Biome>> getNetherBiomes() {
 		return Collections.unmodifiableSet(NETHER_BIOMES);
 	}
 
-	public static Map<Biome, Biome.MixedNoisePoint> getNetherBiomeNoisePoints() {
+	public static Map<RegistryKey<Biome>, Biome.MixedNoisePoint> getNetherBiomeNoisePoints() {
 		return NETHER_BIOME_NOISE_POINTS;
 	}
 
 	private static class DefaultHillsData {
-		private static final ImmutableMap<Biome, Biome> DEFAULT_HILLS;
+		private static final ImmutableMap<RegistryKey<Biome>, RegistryKey<Biome>> DEFAULT_HILLS;
 
-		static WeightedBiomePicker injectDefaultHills(Biome base, WeightedBiomePicker picker) {
-			Biome defaultHill = DEFAULT_HILLS.get(base);
+		static WeightedBiomePicker injectDefaultHills(RegistryKey<Biome> base, WeightedBiomePicker picker) {
+			RegistryKey<Biome> defaultHill = DEFAULT_HILLS.get(base);
 
 			if (defaultHill != null) {
-				picker.addBiome(defaultHill, 1);
-			} else if (BiomeLayers.areSimilar(BuiltinRegistries.BIOME.getRawId(base), BuiltinRegistries.BIOME.getRawId(Biomes.WOODED_BADLANDS_PLATEAU))) {
-				picker.addBiome(Biomes.BADLANDS, 1);
+				picker.addBiome(BuiltinRegistries.BIOME.get(defaultHill), 1);
+			} else if (BiomeLayers.areSimilar(BuiltinRegistries.BIOME.getRawId(BuiltinRegistries.BIOME.get(base)), BuiltinRegistries.BIOME.getRawId(BuiltinRegistries.BIOME.get(Biomes.WOODED_BADLANDS_PLATEAU)))) {
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.BADLANDS), 1);
 			} else if (base == Biomes.DEEP_OCEAN || base == Biomes.DEEP_LUKEWARM_OCEAN || base == Biomes.DEEP_COLD_OCEAN) {
-				picker.addBiome(Biomes.PLAINS, 1);
-				picker.addBiome(Biomes.FOREST, 1);
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.PLAINS), 1);
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.FOREST), 1);
 			} else if (base == Biomes.DEEP_FROZEN_OCEAN) {
 				// Note: Vanilla Deep Frozen Oceans only have a 1/3 chance of having default hills.
 				// This is a clever trick that ensures that when a mod adds hills with a weight of 1, the 1/3 chance is fulfilled.
 				// 0.5 + 1.0 = 1.5, and 0.5 / 1.5 = 1/3.
 
-				picker.addBiome(Biomes.PLAINS, 0.25);
-				picker.addBiome(Biomes.FOREST, 0.25);
+				picker.addBiome(BuiltinRegistries.BIOME.get(Biomes.PLAINS), 0.25);
```

Comment:
![concern](https://cdn.discordapp.com/emojis/707016186281590855.png?v=1) this is not how you use builtin registries

Reply:
How then lol, don't complain and provide no solution.